
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>
  <env name="ROS_LOG_DIR" value="/tmp/"/>

  <arg name="base_frame_id" default="base_link" />
  <arg name="base_footprint_frame_id" default="base_footprint" />
  <arg name="odom_frame_id" default="odom" />
  <arg name="laser_frame_id" default="laser" />
  <arg name="computed_turning_point_frame_id" default="computed_turning_point" />
  <arg name="imu_frame_id" default="base_imu" />
  <arg name="camera_frame_id" default="camera_link" />
  <arg name="controller_frequency" default="1" />

  <arg name="cmd_vel_input_topic" default="/cmd_vel" />
  <arg name="battery_state_output_topic" default="/robot_platform/battery" />
  <arg name="gps_state_output_topic" default="/robot_platform/gps"/>
  <arg name="imu_state_output_topic" default="/robot_platform/imu"/>
  <arg name="odom_state_output_topic" default="/robot_platform/odom"/>
  <arg name="pose_output_topic" default="/robot_platform/pose"/>
  <arg name="wheel_positions_input_topic" default="/robot_platform/wheel_request"/>
  <arg name="platform_status_output_topic" default="/robot_platform/status"/>
  <arg name="shutdown_command_input_topic" default="/robot_platform/shutdown"/>

  <node pkg="robot_platform" type="robot_platform" name="platform_controller" output="screen" respawn="true">

    <param name="serial_dev" type="string" value="/dev/serial/by-id/usb-Teensyduino_USB_Serial_13671670-if00" />
    <param name="serial_baudrate" type="int" value="115200" />
    <param name="controller_frequency" value="$(arg controller_frequency)" />

    <param name="base_frame_id" type="string" value="$(arg base_frame_id)" />
    <param name="odom_frame_id" type="string" value="$(arg odom_frame_id)" />
    <param name="laser_frame_id" type="string" value="$(arg laser_frame_id)" />
    <param name="base_footprint_frame_id" type="string" value="$(arg base_footprint_frame_id)" />
    <param name="computed_turning_point_frame_id" type="string" value="$(arg computed_turning_point_frame_id)" />
    <param name="imu_frame_id" type="string" value="$(arg imu_frame_id)" />
    <param name="camera_frame_id" type="string" value="$(arg camera_frame_id)" />
    
    <param name="cmd_vel_input_topic" type="string" value="$(arg cmd_vel_input_topic)" />
    <!-- <param name="shutdown_command_input_topic" type="string" value="$(arg shutdown_command_input_topic)" /> -->
    
    <param name="battery_state_output_topic" type="string" value="$(arg battery_state_output_topic)" />
    <param name="gps_state_output_topic" type="string" value="$(arg gps_state_output_topic)" />
    <param name="imu_state_output_topic" type="string" value="$(arg imu_state_output_topic)" />
    <param name="platform_status_output_topic" type="string" value="$(arg platform_status_output_topic)" />
    <param name="odometry_output_topic" type="string" value="$(arg odom_state_output_topic)" />
    <param name="pose_output_topic" type="string" value="$(arg pose_output_topic)" />
    
  </node>

</launch>
